storage/src/main/java/io/atomix/storage/buffer/FileBytes.java

```diff
@@ -249,15 +249,15 @@ public Bytes read(int position, Bytes bytes, int offset, int length) {
     if (bytes.hasArray()) {
       try {
         seekToOffset(position);
-        randomAccessFile.read(bytes.array(), (int) offset, (int) length);
+        randomAccessFile.readFully(bytes.array(), (int) offset, (int) length);
```

Comment:
Although RandomAccessFile is not a subclass of InputStream, this method behaves in exactly the same way as the InputStream.read() method of InputStream.
https://docs.oracle.com/javase/7/docs/api/java/io/InputStream.html#read(byte[],%20int,%20int)
An attempt is made to read as many as len bytes, but a smaller number may be read.